File: Cランクレベルアップメニュー/辞書/step2.js
const main = (input) => {
    const lines = input.trim().split('\n');
    const n = Number(lines[0]);
    const names = lines.slice(1, n + 1);
    const m = Number(lines[n + 1]);
    const S = lines[n + m + 2];
    const damages = names.map(name => [name, 0]);
    const objDamages = Object.fromEntries(damages);
    for (let i = n + 2; i < n + m + 2; i++) {
        const [name, damage] = lines[i].split(' ');
        objDamages[name] += Number(damage);
    }
    console.log(objDamages[S]);
 }
 main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: Cランクレベルアップメニュー/辞書/step2.rb
n = gets.to_i
s = {}
n.times do
    name = gets.chomp
    s[name] = 0
end

m = gets.to_i
m.times do
    name, damage = gets.split(' ')
    s[name] += damage.to_i
end

name = gets.chomp

puts s[name]